### PR TITLE
Add ability to provide DNS seeds

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
I'm not sure we can test this in the repo until we implement integration testing. However a few manual test-runs did work with a DNS seed node.

### CODE REVIEW

**Review by TrustHenry — APPROVED** [review]

**Comment by AndrejMitrovic** [comment]
Rebased against v0.x.x.

File: source/agora/node/Network.d [hunk 1]
```diff
@@ -242,3 +249,58 @@ public class NetworkManager
...
+
+    Params:
+        addresses = the set of DNS seeds
+
+    Returns:
+        The resolved set of IPs
+
+*******************************************************************************/
+
+private Set!Address resolveDnsSeeds ( in string[] dns_seeds )
```
Comment (Geod24) [inline source/agora/node/Network.d]:
```suggestion
private Set!Address resolveDNSSeeds (in string[] dns_seeds)
```

**Review by Geod24 — APPROVED** [review]
LGTM, except one minor style fix

**Comment by AndrejMitrovic** [comment]
@Geod24 Fixed and squashed.

**Comment by AndrejMitrovic** [comment]
Btw, I wasn't sure whether we can provide a list of initial nodes and also DNS seeds in the same config ~section~ option. But we could improve upon this later.

**Comment by Geod24** [comment]
> @Geod24 Fixed and squashed.

You didn't modify the call site tho

File: source/agora/node/Network.d [hunk 2]
```diff
@@ -78,12 +81,16 @@ public class NetworkManager
...
+
+        this.dns_seeds = dns_seeds;
     }
 
     /// try to discover the network until we found
     /// all the validator nodes from our quorum set.
     public void discover ()
     {
+        logInfo("Resolving DNS from %s", this.dns_seeds);
+        this.addAddresses(resolveDnsSeeds(this.dns_seeds));
```
Comment (Geod24) [inline source/agora/node/Network.d]: Also what that in a `if (this.dns_seeds.length)`
We don't want to make DNS seed mandatory for the moment, do we ?

**Comment by AndrejMitrovic** [comment]
Rebased.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**